tasks/terraform_apply.rb

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+require_relative '../lib/cli_helper.rb'
+require 'json'
+require 'open3'
+
+class TerraformApply < TaskHelper
+  def apply(opts)
+    dir = File.expand_path(opts[:dir], opts[:_boltdir]) if opts[:dir]
+    cli_opts = %w[-auto-approve -no-color -input=false]
+    cli_opts << "-state=#{File.expand_path(opts[:state], dir)}" if opts[:state]
+
+    if opts[:target]
+      resources = opts[:target].is_a?(Array) ? opts[:target] : Array(opts[:target])
+      resources.each { |resource| cli_opts << "-target=#{resource}" }
+    end
+
+    opts[:var].each { |k, v| cli_opts << "-var '#{k}=#{v}'" } if opts[:var]
```

Comment:
Might be worth documenting somewhere that we don't support single-quotes in any vars. I don't see it being a huge problem but the error message is a bit confusing if you don't know how this works:

```
lucy ~/githubs/bolt (master) $ be bolt plan run test
Starting: plan test
Starting: task terraform::terraform_apply on localhost
Finished: task terraform::terraform_apply with 1 failure in 0.26 sec
Finished: plan test in 0.28 sec
{
  "kind": "bolt/run-failure",
  "msg": "Plan aborted: run_task 'terraform::terraform_apply' failed on 1 nodes",
  "details": {
    "action": "run_task",
    "object": "terraform::terraform_apply",
    "result_set": [
      {
        "target": "localhost",
        "action": "task",
        "object": "terraform::terraform_apply",
        "status": "failure",
        "value": {
          "_error": {
            "kind": "terraform/apply-error",
            "msg": "sh: 1: Syntax error: Unterminated quoted string\n",
            "details": {
            }
          }
        }
      }
    ]
  }
}
```

Reply:
I don't think this is worth trying to find and escape single-quotes.

Reply:
I'm not seeing where the single quoted strings are coming from? Your plan?

Reply:
Yeah - put it all in a gist https://gist.github.com/lucywyman/1f48a2143cf69bc7643609e5cc3dbda0 I think that's enough context but let me know if there's anything missing

Reply:
eek, thats a tough one. I added a note in the readme. There is a work around though i think it should be fine if you put them in a var-file.